packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,178 @@ it, but it does not seem to be justified. For example:
+will not be routed. This is because there has to be a strict priority enforcement,
+and without root Marathon, it is impossible to decide whether AR should use Mesos
+data or not (-> request routing ambiguity). The reverse is possible though -
+failed Mesos cache update will not prevent requests from being routed as long
+as they can be handled *only* using Marathon data.
+
+### Limitations
+Apart from the one mentioned earlier - that `/service` endpoint should be routing
+only to the tasks launched by root Marathon, there is one more limitation that
+needs to be kept in mind.
+
+To properly remove framework, a cleanup procedure needs to be followed:
+* framework needs to be removed from root Marathon
+* it needs to be unregistered with Mesos
+* Zookeeper state has to be cleaned up
+* Mesos reservations (if any), need to be deleted
+
+Without it Mesos framework is still present in the output of `/state-summary`
+endpoint and thus Admin Router will try to route it instead of replying with 404.
+It is impossible for AR to differentiate between frameworks that:
+* have been "just" removed/deleted by Marathon
+* become disconnected from Mesos for some reason, but *MAY* come back at some
+  point
+
+What is more - continuous installation and removal of a framework using the same
+name will cause Mesos to keep entries for them in `/state-summary`. There is no
+enforcement of unique framework names, and thus AR will try to route to one of
+the frameworks (not necessarily the one that is *really* alive) in an undefined
+manner.
+
+Until unique service names are implemented, or a proper cleanup is done by the client,
+a workaround is to launch frameworks always with a different name, for example:
+* `/hello-world-1`
+* `/hello-world-2`
+* `/hello-world-3`
+* `/hello-world-4`
+* etc...
+
+### Recommended way to expose services
+Admin Router allows Marathon tasks to define custom service UI and HTTP
+endpoints, which are made available via `<dcos-cluster>/service/<service-name>`.
+This can be achieved by setting the following Marathon task labels:
 
-The endpoint should only use relative links for links and referenced assets such as .js and .css files. This is due to the fact, that the linked resources will be reachable only in their relative location `<dcos-cluster>/services/<service-name><link>`.
+```
+"labels": { "DCOS_SERVICE_NAME": "service-name", "DCOS_SERVICE_PORT_INDEX": "0", "DCOS_SERVICE_SCHEME": "http" }
+```
 
-Tasks running in nested [Marathon app groups](https://mesosphere.github.io/marathon/docs/application-groups.html) will be available only using their service name (i.e, `<dcos-cluster>/service/<service-name>`) and not considering the marathon app group name (i.e., `<dcos-cluster>/service/app-group/<service-name>`).
+When your container/task has its own IP (typically when running in a virtual
+network), `http://<dcos-cluster>/service/service-name` would be forwarded to the
```

Comment:
s/would be/is

"would be" is appropriate if the sentence started with "If your container had its own"